core/dbt/task/run.py

```diff
@@ -98,7 +99,7 @@ def get_execution_status(sql: str, adapter: BaseAdapter) -> Tuple[RunStatus, str
         return status, message
 
 
-def track_model_run(index, num_nodes, run_model_result):
+def track_model_run(index, num_nodes, run_model_result, adapter=BaseAdapter):
```

Comment:
This is a defensive move in case somehow any other company package relies on this method for telemetry (doubt it, but I'm being safely paranoid ;) )!

Reply:
Fair point that it's possible to not have an adapter, in that case should this be optional? We could handle it as: if `adapter=None` populate `adapter_info` with an empty dict `{}`?

Reply:
I don't think it's possible to not have the base adapter right? So you'd get the in-scope base adapter version which right now would include (1) name (2) version (3) an empty dict for adapter-specific details

Does that make sense or would you prefer None? I actually thought about None, but that felt like an illegal state to reach this code in

Reply:
I can agree with returning a default result of `{}` in the no adapter scenario. That's sufficiently defensive to quell my anxiety.